mapstruct-2/src/main/java/com/baeldung/mapstruct/date/mapper/ReuseUserMapper.java

```diff
@@ -0,0 +1,11 @@
+package com.baeldung.mapstruct.date.mapper;
+
+import com.baeldung.mapstruct.date.model.User;
+import com.baeldung.mapstruct.date.model.UserDTO;
+import org.mapstruct.Mapper;
+
+@Mapper(uses = DateMapper.class)
+public interface ReuseUserMapper {
+
+    User toUser(UserDTO userDTO);
```

Comment:
Why is there no `@Mapping` here? How does mapstruct know which field to map, is it just by field name matching exactly? I think we should make this explicit to avoid confusion for the readers

Reply:
This is for section 5 of the article to demonstrate the idea of resusing general conversion method. Mapstruct convert field by field name matching, if no extra attribute need to be set, mapping annotation is not required.

Reply:
I think we should stay on topic and only vary the parts that are relevant to this specific article. I don't want the reader to have to understand this additional concept that isn't part of the intended topic scope. So let's use `@Mapping` consistently across the entire article please.

Reply:
@Mapping is one of the ways to convert string to date. Using @Mapper is another way to do so. I shared the outline and you confirmed its okay.

Reply:
Either way, we should make sure the field mapping is explicit. Is this possible with `@Mapper`? I can't let this pass until we have a way to explicitly specify the field name

Reply:
My understanding is that `@Mapper(uses = DateMapper.class)` is like an import for another mapper. Perhaps this means `DateMapper` should have `@Mapping`?

There must be a way to specify field names with this approach and it can be a very critical thing - we don't always have class field names that match the names of the fields we want to match 1:1

Reply:
Perhaps it's better to use an example where the field name doesn't match so we can clarify how that is mapped in both of the use cases you are covering

Reply:
`public Date mapStringToDate(String date) ` gives us information on how to map from a String to a Date, like a library method. It doesn't tell us how to map the fields.

We should be using both `@Mapping` and `@Mapper` here, they cover totally separate concerns